src/recorder/recorder_worker_hdf5.cc

```diff
@@ -0,0 +1,313 @@
+/**
+ * @file recorder_worker_multifile.cc
+ * @brief Recorder worker to write to multiple bin files per rx symbol
+ */
+
+#include "recorder_worker_hdf5.h"
+
+#include <ctime>
+#include <fstream>
+#include <nlohmann/json.hpp>
+
+#include "logger.h"
+#include "utils.h"
+#include "version_config.h"
+
+namespace Agora_recorder {
+
+static constexpr bool kDebugPrint = false;
+static const std::string kHdf5Filename = "UeRxData.h5";
+static constexpr size_t kFrameInc = 2000;
+static constexpr size_t kDsDimSymbol = 2;
+static constexpr ssize_t kFixedDimensions = -1;
+
+RecorderWorkerHDF5::RecorderWorkerHDF5(const Config* in_cfg,
+                                       size_t antenna_offset,
+                                       size_t num_antennas,
+                                       size_t record_interval)
+    : RecorderWorker(in_cfg, antenna_offset, num_antennas, record_interval),
+      cfg_(in_cfg),
+      antenna_offset_(antenna_offset),
+      num_antennas_(num_antennas),
+      interval_(record_interval),
+      max_frame_number_(0),
+      data_chunk_dims_{
+          {1, 1, 1, 1, static_cast<hsize_t>(2 * cfg_->SampsPerSymbol())}} {}
+
+RecorderWorkerHDF5::~RecorderWorkerHDF5() = default;
+
+void RecorderWorkerHDF5::Init() {
+  AGORA_LOG_INFO("RecorderWorkerHDF5::Creating output HD5F file: %s\n",
+                 kHdf5Filename.c_str());
+  hdf5_ = std::make_unique<Hdf5Lib>(kHdf5Filename, "Data");
+
+  hdf5_->WriteAttribute("SOFTWARE", "AGORA");
+  hdf5_->WriteAttribute("SOFTWARE_VERSION", GetAgoraProjectVersion());
+
+  auto time = std::time(nullptr);
+  auto local_time = *std::localtime(&time);
+
+  std::ostringstream oss;
+  oss << std::put_time(&local_time, "%d-%m-%Y %H-%M-%S");
+  hdf5_->WriteAttribute("TIME", oss.str());
+
+  // Write the entire config
+  const std::string conf_filename = cfg_->ConfigFilename();
+  std::ifstream conf(conf_filename);
+  nlohmann::json data = nlohmann::json::parse(conf, nullptr, true, true);
+  conf.close();
+  hdf5_->WriteAttribute("CONFIG", data.dump());
+
+  // Write Atrributes
+  // ******* COMMON ******** //
+  // TX/RX Frequencyfile
```

Comment:
Are these duplicate after writing the entire config?

Reply:
Yes and No.   Any value not written to the config is generated by defaults based on the program.  So some critical values could be computed in the Config class even though they are not explicit in the config.  This attribute value could be useful to rerun the experiment but I won't be parsing the string just yet.